input.go

```diff
@@ -266,6 +267,15 @@ func EditDynamicWithCallback(defval, prompt string, refresh func(int, int), call
 				bufpos = forwardWordIndex(buffer, bufpos)
 				cursor = termutil.RunewidthStr(buffer[:bufpos])
 			}
+		case "C-i", "TAB":
+			// Allow tabs to be input only for commands which have no need for
+			// tab completion.
+			if prompt == "Search" || prompt == "Find" || prompt == "Find regexp" ||
```

Comment:
Matching against the prompt is a little hairy. These are subject to change (we may also want to localize Gomacs at some point). Unfortunately, the only solution I can think of is a boolean flag on the function call which would be called `allowTabs` or something. We could copy this code to a new function and replace search/find/find-regex with calls to that function, and EditDynamicWithCallback would be an interface that calls the new function with the flag being false. It's not a pretty refactor, but it would work and wouldn't require string comparison.

Reply:
Question: does `C-q C-i` insert a tab?

Reply:
> Matching against the prompt is a little hairy. These are subject to change (we may also want to localize Gomacs at some point).

This could be made somewhat less fragile by defining constants for the prompts and using them in both nav.go and input.go, e.g.:

```go
const (
	PromptSearch = "Search"
	PromptFind = "Find"
	PromptFindRegexp = PromptFind + " regexp"
	PromptReplacePrefix = "Replace "
	PromptReplaceSuffix = " with"
)
```

If localization ever happens, these constants would be replaced by variables.

> Unfortunately, the only solution I can think of is a boolean flag on the function call which would be called `allowTabs` or something. We could copy this code to a new function and replace search/find/find-regex with calls to that function, and EditDynamicWithCallback would be an interface that calls the new function with the flag being false. It's not a pretty refactor, but it would work and wouldn't require string comparison.

Yeah, that's probably more robust.  I can submit a new PR that does it that way.

Reply:
> Question: does `C-q C-i` insert a tab?

With this PR: yes.  Without this PR: no.

Reply:
> I can submit a new PR that does it that way.

Yes, that'd be great; or feel free to add commits to this one, either way.